viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
+        
+        for i in range(len(user_dict["watched"])):
+            genre = user_dict["watched"][i]["genre"]
+            if genre in genre_count:
+                genre_count[genre] += 1
+            else:
+                genre_count[genre] = 1
+        return max(genre_count, key=lambda key:genre_count[key])
+    else:
+        return None
+    
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_dict):
+    friends_watched_list = user_dict["friends"]
+    combined_friends_list = []
+    for watched_dict in friends_watched_list:
+        combined_friends_list.extend(watched_dict["watched"])
+    users_watched_list = user_dict["watched"]
+    unique_watched = []
+    for movie in users_watched_list:
+        if movie not in combined_friends_list:
+            unique_watched.append(movie)
+    return unique_watched
+
+
+def get_friends_unique_watched(user_dict):
+    friends_watched_list = user_dict["friends"]
+    combined_friends_list = []
+    for watched_dict in friends_watched_list:
+        combined_friends_list.extend(watched_dict["watched"])
+    users_watched_list = user_dict["watched"]
+    unique_watched = []
+    for movie in combined_friends_list:
+        if movie not in users_watched_list:
+            if movie not in unique_watched:
+                unique_watched.append(movie)
+    return unique_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_dict):
+    recommendations = []
+    friends_unique_watched = get_friends_unique_watched(user_dict)
+    for movie in friends_unique_watched:
+        if movie["host"] in user_dict["subscriptions"]:
+            recommendations.append(movie)
+    return recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_dict):
+    friends_list = user_dict["friends"]
+    for friend in friends_list:
+        if not friend["watched"]:
+            return []
+    if user_dict["watched"]:
+        recommendations = []
+        watched = user_dict["watched"]
+        subscriptions = user_dict["subscriptions"]
+        fave_genres = []
+        for movie in user_dict["favorites"]:
+            if movie["genre"] not in fave_genres:
+                fave_genres.append(movie["genre"])
+        friends_unique_watched = get_friends_unique_watched(user_dict)
+        for movie in friends_unique_watched:
+            if (movie not in watched) \
+                and (movie["host"] in subscriptions) \
+                and (movie["genre"] in fave_genres):
+
+                recommendations.append(movie)
+        return recommendations
+    else:
+        return []
+
+
+
+def get_rec_from_favorites(user_dict):

```

Comment:
👍 

Though consider starting from the unique user watched data and filtering it by whether each film is present in their favorite list. This _does_ requires us to assume that any movie in the user's favorite list should also have been one they watched, but it would let us leverage our `get_unique_watched` function.

We could also consider building a set of titles here to help us perform those `in` checks.